nats/server_report_command.go

```diff
@@ -472,31 +492,92 @@ func (c *SrvReportCmd) renderConnections(report []*server.ConnInfo) {
 		iBytes += info.InBytes
 		subs += info.NumSubs
 
-		table.AddRow(info.Cid, name, info.IP, info.Account, info.Uptime, humanize.Comma(info.InMsgs), humanize.Comma(info.OutMsgs), humanize.IBytes(uint64(info.InBytes)), humanize.IBytes(uint64(info.OutBytes)), len(info.Subs))
+		srvName := info.Info.Name
+		cluster := info.Info.Cluster
+
+		acc := info.Account[0:10] + ".." + info.Account[46:]
+
+		table.AddRow(info.Cid, name, srvName, cluster, info.IP, acc, info.Uptime, humanize.Comma(info.InMsgs), humanize.Comma(info.OutMsgs), humanize.IBytes(uint64(info.InBytes)), humanize.IBytes(uint64(info.OutBytes)), len(info.Subs))
 	}
 
 	if len(report) > 1 {
 		table.AddSeparator()
-		table.AddRow("", "", "", "", "", humanize.Comma(iMsgs), humanize.Comma(oMsgs), humanize.IBytes(uint64(iBytes)), humanize.IBytes(uint64(oBytes)), humanize.Comma(int64(subs)))
+		table.AddRow("", "", "", "", "", "", "", humanize.Comma(iMsgs), humanize.Comma(oMsgs), humanize.IBytes(uint64(iBytes)), humanize.IBytes(uint64(oBytes)), humanize.Comma(int64(subs)))
 	}
 
 	fmt.Print(table.Render())
 }
 
+type connz struct {
+	*server.Connz
+	*server.ServerInfo
+}
+
+type connzList []connz
+
+func (c connzList) flatConnInfo() []connInfo {
+	var conns []connInfo
+	for _, conn := range c {
+		for _, c := range conn.Conns {
+			conns = append(conns, connInfo{c, conn.ServerInfo})
+		}
+	}
+	return conns
+}
+
+func parseConnzResp(resp []byte) (*connz, error) {
+	reqresp := map[string]json.RawMessage{}
+
+	err := json.Unmarshal(resp, &reqresp)
+	if err != nil {
+		return nil, err
+	}
+
+	errresp, ok := reqresp["error"]
+	if ok {
+		return nil, fmt.Errorf("invalid response received: %#v", errresp)
+	}
+
+	data, ok := reqresp["data"]
+	if !ok {
+		return nil, fmt.Errorf("no data received in response: %#v", reqresp)
+	}
+
+	c := connz{
+		Connz:      &server.Connz{},
+		ServerInfo: &server.ServerInfo{},
+	}
+
+	s, ok := reqresp["server"]
+	if !ok {
+		return nil, fmt.Errorf("no server data received in response: %#v", reqresp)
+	}
```

Comment:
Have to say I find the way these data structures are designed so user unfriendly that they are nearly unusable